File: Code/Aula_07/Ex4.c
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>
#include <unistd.h>

void funcao_para_control_c()
{
	printf("\nQuem mandou voce pressionar CTRL-C?\n");
	printf("Vou ter de fechar o programa!\n");
	exit(1);
}

int main()
{
	signal(SIGINT, funcao_para_control_c);
	printf("Pressione CTRL-C para retirar o programa do loop infinito abaixo.\n");
	while(1);
	return 0;
}

File: 2_Programação_sistemas_operacionais/2.4_Pipes_Sinais_Alarmes/Python/Ex4.py
import signal

def funcao_para_control_c(signum, frame):
	print("\nQuem mandou você pressionar CTRL-C?")
	print("Vou ter de fechar o programa!")
	exit(1)

signal.signal(signal.SIGINT, funcao_para_control_c)
print("Pressione CTRL-C para retirar o programa do loop infinito abaixo.")
while True:
	pass